src/cocaine-app/jobs/restore_group.py

```diff
@@ -88,9 +88,14 @@ def create_tasks(self):
 
         self.check_node_backends(src_group)
 
-        old_group_state = infrastructure.get_group_history(group.group_id)['nodes'][-1]['set']
-        if old_group_state:
-            old_host, old_port, old_family, old_backend_id, old_base_path = old_group_state[0][:5]
+        old_group_node_backends_set = infrastructure.get_group_history(group.group_id).nodes[-1].set
+        if old_group_node_backends_set:
+            old_node_backend = old_group_node_backends_set[0]
+            old_host = cache.get_ip_address_by_host(old_node_backend.hostname)
+            old_port = old_node_backend.port
+            old_family = old_node_backend.family
+            old_backend_id = old_node_backend.backend_id
+            old_base_path = old_node_backend.path
```

Comment:
This is too complex.
Let me do it in https://github.com/yandex/mastermind/issues/22